arithmetization/src/main/java/net/consensys/linea/zktracer/module/rlpaddr/RlpAddr.java

```diff
@@ -109,7 +109,13 @@ public void callRlpAddrCreate2(MessageFrame frame, Bytes32 salt, Bytes32 hash) {
     final Address currentAddress = frame.getRecipientAddress();
     final Bytes32 rawCreate2Address = getCreate2RawAddress(currentAddress, salt, hash);
     final RlpAddrOperation operation =
-        new RlpAddrOperation(rawCreate2Address, OpCode.CREATE2, currentAddress, salt, hash);
+        new RlpAddrOperation(
+            rawCreate2Address,
+            OpCode.CREATE2,
+            currentAddress,
+            salt,
+            hash,
+            longToUnsignedBigInteger(frame.getWorldUpdater().get(currentAddress).getNonce()));
```

Comment:
Even though it's unnecessary, the nonce is part of the HUB -> RLPADDR lookup for CREATE2.